https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/

Specific key steps include:

- (1)
    
    Freezing ribosomes on mRNAs to avoid ribosome run-off: After culturing the cells, they are treated with translation inhibitors [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)]. Even though there are a wide variety of translation inhibitors (such as ribosome-targeting initiation, elongation, and termination inhibitors; inhibitors of initiation and elongation factors; and the inhibitors of cell signaling pathways involved in translational control), arguably the most commonly used for ribosomal footprinting are the classical elongation inhibitors, such as cycloheximide, anisomycin, emetine [[7](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B7-cells-11-02966)]. These inhibitors, however, have to be used with caution, considering their potential artifact effects in ribosomal distribution, which are specific to cell/organism type, cell growth conditions, and the contents of lysis buffer [[8](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B8-cells-11-02966)]. For instance, treatment with cycloheximide results in a larger ribosomal distribution at the 5′ coding end of the transcript. The effect is more pronounced in S. Cerevisiae and in shorter transcripts [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966),[7](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B7-cells-11-02966),[8](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B8-cells-11-02966)]. Alternatively, if the goal is to specifically inhibit ribosomes at their initiation sites, inhibitors specific to de novo initiation ribosomes are used: harringtonine and lactimidomycin are examples of inhibitors with such activity [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966),[7](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B7-cells-11-02966)]. In light of potential challenges associated with the use of translation inhibitors, in situ detergent lysis of cells is advised where applicable [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)]. Flash-freezing and cryogenic pulverization may be suitable for specific cell and tissue types (mainly animal cells and tissues) [[2](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B2-cells-11-02966),[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966),[8](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B8-cells-11-02966)].
    
- (2)
    
    Ribonuclease digestion breaks down unprotected regions of mRNA: Ribonucleases are responsible for the breaking down and degradation of exposed mRNA, except for ribosome-protected fragments (RPFs). The precision of this step is dependent on buffer conditions, where a lower concentration of sodium (150–200 mM) and magnesium (5mM-10mM) is preferred for optimally uniform digestion by the nuclease [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)].
    
- (3)
    
    mRNA–ribosome complex recovery and RNA purification: After ribonuclease digestion, the mRNA–ribosome footprint complexes are recovered using sucrose density fractionation or sedimentation through a sucrose cushion [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)]. Alternatively, mRNA–ribosome complexes can be retrieved by gel filtration [[9](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B9-cells-11-02966)] or affinity purification with the use of an epitope tag added to the ribosomes, which is essential in profiling specific cell types [[10](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B10-cells-11-02966),[11](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B11-cells-11-02966)]. The resultant ribosomal pellet is resuspended in Trizol/Qiazol reagent, and the protected RNA fragments are then purified using a spin column, such as miRNeasy kit [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)]. Subsequently, the isolated RNA is separated by polyacrylamide gel electrophoresis to isolate the RNA fragments of 26–34 nt length corresponding to the RPFs.
    
- (4)
    
    Library generation and deep sequencing: Following RPF purification, it is essential to deplete the sample of rRNA. Ribosomal RNA is the most predominant RNA, making up about 80% of cellular RNA in the cell, and its removal is essential because the abundance of rRNA leads to fewer RPFs being sequenced and fewer mapping reads. This, as a result, reduces the useful size of the library. rRNA can be depleted using kits, such as Ribo-zero Plus or Legacy Ribo-zero (Illumina, San-Diego, C), which use anti-rRNA DNA probes with subsequent removal of rRNA by enzymatic digestion (RNase H) or by streptavidin affinity pull-down assay. When RPFs are depleted of rRNA, the fragments need to be ligated or fused to adapters [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)]. RPFs can be tailed with a poly-A polymerase, or else a single-stranded RNA linker of a known sequence can be ligated to the 3′-end of the RPF [[12](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B12-cells-11-02966)]. The RNA fragments are then converted into complementary DNA by the enzyme reverse transcriptase. The cDNA later serves as a template for exponential amplification using PCR. This library can then be sequenced to generate a bioinformatics library containing RPF sequenced reads [[2](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B2-cells-11-02966),[13](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B13-cells-11-02966)]. As an alternative strategy, rRNA depletion can be carried out after the reverse-transcription-step cDNA is circularized. The rRNA subtraction oligos (biotinylated sense-strand oligonucleotides against rRNA-derived cDNA) are added. rRNA-derived cDNA fragments are removed using a streptavidin affinity assay. The remaining cDNA fragments represent the non-rRNA RPFs, which are further PCR-amplified and sequenced [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)]. Mapping these sequenced RPFs to the transcriptome provides a snapshot of translation that reveals transcriptome wide the positions and densities of ribosomes on individual mRNAs.
    
- (5)
    
    Data analysis: The upstream steps of Ribo-seq analysis are similar to the standard RNA-seq data analysis workflow [[14](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B14-cells-11-02966)], including read mapping to transcripts. The downstream Ribo-seq data analysis includes calculating transcript translational efficiency when the data are paired with RNA-seq, identifying differentially translated transcripts, de novo identifying transcript open reading frame (ORF), and detecting potential translational pausing events [[6](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9562884/#B6-cells-11-02966)].